src/sema/types.rs

```diff
@@ -1684,8 +1692,12 @@ impl Type {
                 Type::Value => BigInt::from(ns.value_length),
                 Type::Uint(n) | Type::Int(n) => BigInt::from(n / 8),
                 Type::Rational => unreachable!(),
-                Type::Array(_, dims) if dims.first() == Some(&ArrayLength::Dynamic) => {
-                    BigInt::from(4)
+                Type::Array(_, dims) if dims.contains(&ArrayLength::Dynamic) => {
+                    let mut size = dynamic_array_size(dims);
+
+                    // A pointer is four bytes on Solana
+                    size.mul_assign(4);
+                    size
```

Comment:
size + 4?

Reply:
> size + 4?

`size * 4`, right? The operation is `size.mul_assign(4)`.